src/trix/models/composition.coffee

```diff
@@ -109,8 +120,12 @@ class Trix.Composition extends Trix.BasicObject
       else
         if block.isEmpty()
           @removeLastBlockAttribute()
-        else if block.text.getStringAtRange([endLocation.offset - 1, endLocation.offset]) is "\n"
+        else if previousCharacter is "\n"
+          @breakFormattedBlock()
+        else if breaksOnReturn and nextCharacter is "\n"
           @breakFormattedBlock()
+        else if breaksOnReturn and previousCharacter is ""
```

Comment:
Checking `previousCharacter is ""` makes it seem like an empty string is somehow significant, but they never exist in a document. I believe this catches `startLocation.offset is 0` (beginning of block), yeah?